File: auth/php/newtoken.php
#!/usr/bin/env php
<?php
require 'vendor/autoload.php';

$deviceId = getenv('DEVICE_ID');
$jwtIssuer = getenv('JWT_ISSUER');
$jwtSubject = getenv('JWT_SUBJECT');
$jwtSecret = getenv('JWT_SECRET');

// Example based on lcobucci/jwt v3.3 (latest stable at time of writing).
use Lcobucci\JWT;

$signer = new JWT\Signer\Hmac\Sha256();
$key = new JWT\Signer\Key($jwtSecret);

$now = time();
$token = (new JWT\Builder())->issuedBy($jwtIssuer)
                            ->relatedTo($jwtSubject)
                            ->expiresAt($now + 30*60)
                            ->withClaim('grants', array('connect:device:' . $deviceId))
                            ->getToken($signer, $key);

echo $token . "\n";

?>

File: auth/go/newtoken.go
package main

import (
	"fmt"
	"os"
	"time"

	"github.com/dgrijalva/jwt-go"
)

var (
	deviceID   = os.Getenv("DEVICE_ID")
	jwtIssuer  = os.Getenv("JWT_ISSUER")
	jwtSubject = os.Getenv("JWT_SUBJECT")
	jwtSecret  = []byte(os.Getenv("JWT_SECRET"))
)

func main() {
	signingMethod := jwt.SigningMethodHS256
	claims := jwt.MapClaims{
		"iss":    jwtIssuer,
		"sub":    jwtSubject,
		"exp":    time.Now().Add(30 * time.Minute).Unix(),
		"grants": []string{"connect:device:" + deviceID},
	}
	token, err := jwt.NewWithClaims(signingMethod, claims).SignedString(jwtSecret)
	if err != nil {
		panic(err)
	}
	fmt.Println(token)
}
